### PR TITLE
feat: pass markdown for hint message on field

### LINKED ISSUE #5559: Support markdown in widget hint texts
I'd like to make widget help texts more useful by adding links to external documentation, and using bold and italics for emphasis. 

**Describe the solution you'd like**
I'd like the ```hint``` option to support a subset of markdown so I can add links, bold, and italics. Alternatively, I'd also be okay with writing straight HTML to do the same thing.

At the moment neither of these options work: Markdown isn't processed and HTML is escaped by React:

image

Thanks!

### PR DESCRIPTION
**Pull request by gdgkirkley**
**Summary**
This adds the option to include HTML in the hint message on a field. Closes #5559 

The issue creator requested support for Markdown, which could be added in future but would require parsing the markdown from the config and would be a larger change. This is a simpler addition that addresses the same issue.

**Test plan**
I didn't see any existing tests for these components, but would be more than happy to add if I just missed them! This passes the existing tests.

image

**A picture of a cute animal (not mandatory but encouraged)**
Cute pup

### CODE REVIEW

File: dev-test/config.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ collections: # A list of collections the CMS should be able to edit
         required: false
         tagname: ''
 
-      - { label: 'Body', name: 'body', widget: 'markdown', hint: 'Main content goes here.' }
+      - { label: 'Body', name: 'body', widget: 'markdown', hint: 'Main content goes here. <a href="http://example.com">Can include HTML!</a>' }
```
Comment (erezrokah) [inline dev-test/config.yml:53]: Can we revert this change?

File: packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js [hunk 2]
```diff
@@ -327,9 +331,11 @@ class EditorControl extends React.Component {
               isFieldHidden={isFieldHidden}
             />
             {fieldHint && (
-              <ControlHint active={isSelected || this.state.styleActive} error={hasErrors}>
-                {fieldHint}
-              </ControlHint>
+              <ControlHint
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:334]: I believe the intention in the issue is to support markdown, not injecting HTML directly (which we would like to avoid).

We could use https://github.com/remarkjs/react-markdown or even [expose the one we're using for the markdown widget preview](https://github.com/netlify/netlify-cms/blob/574025fd649d49f70650a6112617642a6cb24f07/packages/netlify-cms-widget-markdown/src/MarkdownPreview.js#L20)

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @gdgkirkley, can you look into supporting markdown and not HTML directly?

File: dev-test/config.yml (hunk 1 above)
Comment (gdgkirkley) [reply dev-test/config.yml:53]: Yep!

File: packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js (hunk 2 above)
Comment (gdgkirkley) [reply packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:334]: Ah cool - sounds good! I will update to use one or the other with markdown.

**Comment by gdgkirkley** [comment]
Hey @erezrokah - I've updated to use `react-markdown` and removed that change to the dev config. `react-markdown` seemed like a great option since it can limit the allowed elements to just the bold, itatlic and link tags requested in the issue.

I also made an update to the docs for this change - hope that makes sense for this change!

File: packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js [hunk 3]
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
             />
             {fieldHint && (
               <ControlHint active={isSelected || this.state.styleActive} error={hasErrors}>
-                {fieldHint}
+                <ReactMarkdown
+                  allowedElements={['a', 'strong', 'em']}
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:333]: `em` should be used to support italic.

Also, with `unwrapDisallowed=true` we don't need the `p` element or the added styling.

File: packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js [hunk 4]
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
...
+                  unwrapDisallowed={true}
+                  components={{
+                    // eslint-disable-next-line no-unused-vars
+                    a: ({ node, ...props }) => (
+                      <a
+                        {...props}
+                        target="_blank"
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:340]: Open links in a new tab

File: packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js [hunk 5]
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
...
+                        rel="noopener noreferrer"
+                        style={{ color: 'inherit' }}
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:342]: Inherit the style for links

**Review by erezrokah — APPROVED** [review]
Thanks for the follow up @gdgkirkley!

I've made some minor changes to fix italic, open links in a new tabs and also apply the active/error style (active - when the field is focused, error when for example the field is required and the user clicks save).

image

image
